### PR TITLE
No more os detection

### LINKED ISSUE #38: OS detection is wrong
You can't just assume `__os="Linux"` (https://github.com/kvz/bash3boilerplate/blob/1f2e1842b3c7158b291563fbb7626ee14a7ee1e5/main.sh#L29)

Suggestion: use `uname(1)` instead.

### PR DESCRIPTION
**Pull request by kvz**
Merging this fixes #38

### CODE REVIEW

**Comment by zbeekman** [comment]
LGTM. I think L12 addresses my comment about making the user aware of `$OSTYPE` sufficiently :+1:

**Comment by kvz** [comment]
Yes indeed, you mean https://github.com/kvz/bash3boilerplate/blob/no-more-os-detection/main.sh#L233 correct?

Anyway, I'm glad you like it. Shipping it is then 😄

File: main.sh [hunk 1]
```diff
@@ -245,7 +230,7 @@ fi
 info "__file: ${__file}"
 info "__dir: ${__dir}"
 info "__base: ${__base}"
-info "__os: ${__os}"
+info "OSTYPE: ${OSTYPE}"
```
Comment (zbeekman) [inline main.sh]: This is the line I'm talking about. My Github iPhone  app may have botched the line numbers.

Comment (kvz) [inline main.sh]: Yes, talking about the same thing then, perfect! 👌

**Pull request merged**